### PR TITLE
Implement retina-aware image context creation

### PR DESCRIPTION
**Pull request by jwilling**
Also, I allowed for the passing in of `0` to base the scale on the current screen's scale, just like on iOS.

### CODE REVIEW

File: lib/UIKit/TUICGAdditions.m [hunk 1]
```diff
@@ -199,24 +199,31 @@ void TUIGraphicsPopContext(void)
...
 	CGImageRelease(CGImage);
-
+    
 	return image;
 }
 
 void TUIGraphicsBeginImageContextWithOptions(CGSize size, BOOL opaque, CGFloat scale)
 {
+    // as in the iOS docs, "if you specify a value of 0.0, the scale
+    // factor is set to the scale factor of the device’s main screen."
```
Comment (jspahrsummers) [inline lib/UIKit/TUICGAdditions.m]: Do you mind adding this to the header instead?

**Comment by jspahrsummers** [comment]
Can you convert spaces to tabs in these changes? After that (and the other note above), :+1:.

**Comment by jwilling** [comment]
Heh, didn't even notice the tabs. Any reason why tabs are used over spaces?

**Comment by jspahrsummers** [comment]
That debate has been raging since people have been coding. Two reasons on this project:
1. Convention.
2. Tabs are characters meant for indentation. :P

Anyways, thanks for submitting this! :gem:

**Comment by jwilling** [comment]
:thumbsup:

**Pull request merged**